addons/website/controllers/main.py

```diff
@@ -222,7 +222,14 @@ def robots(self, **kwargs):
         # Don't use `request.website.domain` here, the template is in charge of
         # detecting if the current URL is the domain one and add a `Disallow: /`
         # if it's not the case to prevent the crawler to continue.
-        return request.render('website.robots', {'url_root': request.httprequest.url_root}, mimetype='text/plain')
+        allowed_routes = self._get_allowed_robots_routes()
+        content = request.env['ir.ui.view']._render_template(
+            'website.robots',
+            {
+                'url_root': request.httprequest.url_root,
+                'allowed_routes': allowed_routes,
+            })
```

Comment:
For the sake of fixing this without requiring everyone who uses the app to update I guess we should go back to appending text to this.

`content + '\n' + '\n'.join(allowed_routes)`

And we can go back to the "cleaner" implementation in master